blog/2022-11-01-cfd-protocol-explained.md

```diff
@@ -0,0 +1,247 @@
+
+Importantly, the `refund_transaction` works the same as before: it spends the `dlc_output`, which is now an output to the `commitment_transaction`.
+We focus on the CETs for clarity, but everything applies to the `refund_transaction` as was discussed in previous sections.
+
+As an example, we can imagine maker and taker agreeing to roll over the CFD because they're both optimistic that the price movement of BTC will benefit them if they wait longer.
+During `contract_setup`, they created and signed `commitment_transaction_0` and a whole set of CETs along with it.
+During `rollover`, they produce a new `commitment_transaction_1` and a new collection of CETs based on a new `announcement` from `olivia`.
+Since they have the means to publish fully signed `commitment_transaction`s, they can eventually publish `commitment_transaction_1` and settle the CFD based on a newer `attestation` from `olivia`.
+
+##### Punish transaction
+
+But, what happens if either party decides to publish an old `commitment_transaction_0` and settle the CFD based on an old BTC price attested by the oracle?
+That would defeat the purpose of rolling over the CFD.
+This is where the revocation mechanism comes in.
+
+A `commitment_transaction`'s `dlc_output` can actually be spent in one more way.
+In full, the `commitment_transaction` can be spent with:
+
+    1. A series of CETs unlocked by one of `olivia`'s `attestation`s.
+    2. A `refund_transaction` that can only be published at least some time after `olivia` has published the relevant `attestation`.
+    3. One `punishment_transaction` per party, which can be constructed unilaterally if one's counterparty publishes a revoked `commitment_transaction`.
+    This transaction allows the aggrieved party to claim all the coins in the `dlc_output`.
+
+During `contract_setup` and every subsequent `rollover`, parties work together to add another spending condition to the `dlc_output`.
+They each generate a revocation keypair `(r, R)` and a publishing keypair `(p, P)` and share the public keys with each other.
+The `dlc_output` is no longer a simple 2-of-2 multisig on their public keys `A` and `B`.
+It also includes:
+
+    1. A way for the maker to spend it with knowledge of their own secret key `a`, `r_taker` and `p_taker`.
+    1. A way for the taker to spend it with knowledge of their own secret key `b`, `r_maker` and `p_maker`.
+
+Instead of exchanging regular signatures on the `commitment_transaction`, they create one adaptor signature each.
+For example, the taker encrypts under `P_maker` their signature for their public key `B` on the `commitment_transaction`.
+Later on, the maker can decrypt the taker's adaptor signature using the publishing secret key `p_maker`.
+If they eventually publish the `commitment_transaction` using the decrypted signature, they leak `p_maker` to the taker.
+
+Additionally, every time after a contract `rollover` is completed, both parties must exchange their revocation _secret_ keys.
```

Comment:
Should we mention the edge case where one party does not share the revocation secrets?

Reply:
Good point. I'll add something about that!